appsec/src/extension/request_abort.c

```diff
@@ -601,13 +626,31 @@ void dd_request_abort_startup()
     _content_type_json_zstr =
         zend_string_init_interned(ZEND_STRL(JSON_CONTENT_TYPE), 1);
 
+#ifdef FRANKENPHP_SUPPORT
+    if (strcmp(sapi_module.name, "frankenphp") == 0) {
+        // prepare a noop zend_compile_file for req init blocking in frankenphp
+        _orig_zend_compile_file = zend_compile_file;
+        zend_compile_file = _req_init_block_zend_compile_file;
+    }
+#endif
```

Comment:
This needs to happen in post_startup otherwise it won't be executed with opcache active.

Reply:
Thanks, fixed in [8355494](https://github.com/DataDog/dd-trace-php/pull/3165/commits/8355494de338c5c9f0c5f0d61b3e686c719048d3).